src/Sdk/Cerbos/Sdk/CerbosClient.cs

```diff
@@ -0,0 +1,82 @@
+// Copyright 2021-2023 Zenauth Ltd.
+// SPDX-License-Identifier: Apache-2.0
+
+using System;
+using System.Threading.Tasks;
+using Cerbos.Sdk.Response;
+
+namespace Cerbos.Sdk
+{
+    /// <summary>
+    /// CerbosClient provides a client implementation that communicates with the PDP.
+    /// </summary>
+    public sealed class CerbosClient
+    {
+        private Api.V1.Svc.CerbosService.CerbosServiceClient CerbosServiceClient { get; }
+
+        public CerbosClient(Api.V1.Svc.CerbosService.CerbosServiceClient cerbosServiceClient)
+        {
+            CerbosServiceClient = cerbosServiceClient;
+        }
+
+        /// <summary>
+        /// Send a request consisting of a principal, resource(s) & action(s) to see if the principal is authorized to do the action(s) on the resource(s).
+        /// </summary>
+        public CheckResourcesResponse CheckResources(Builder.CheckResourcesRequest request)
+        {
+            try
+            {
+                return new CheckResourcesResponse(CerbosServiceClient.CheckResources(request.ToCheckResourcesRequest()));
+            }
+            catch (Exception e)
+            {
+                throw new Exception($"Failed to check resources: ${e}");
+            }
+        }
+        
+        /// <summary>
+        /// Send an async request consisting of a principal, resource(s) & action(s) to see if the principal is authorized to do the action(s) on the resource(s).
+        /// </summary>
+        public async Task<CheckResourcesResponse> CheckResourcesAsync(Builder.CheckResourcesRequest request)
+        {
+            try
+            {
+                return new CheckResourcesResponse(await CerbosServiceClient.CheckResourcesAsync(request.ToCheckResourcesRequest()));
```

Comment:
I am not familiar with C# async. Shouldn't this be something like the following? (Probably not valid C#)

Calling `await` typically blocks in other languages so I am not sure whether we should do that within the SDK. It should be up to the user.

```suggestion
                return CerbosServiceClient.CheckResourcesAsync(request.ToCheckResourcesRequest()).ContinueWith(r => new CheckResourcesResponse(r.Result));
```